projects/knora-ui/src/lib/viewer/values/time-value/time-value.component.ts

```diff
@@ -0,0 +1,198 @@
+import { BaseValueComponent } from '..';
+import { FormControl, FormGroup, FormBuilder, ValidatorFn, AbstractControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { IntervalErrorStateMatcher } from '../interval-value/interval-value.component';
+import { DatePipe } from '@angular/common';
+import * as _ from 'lodash';
+import { CustomRegex } from '../custom-regex';
+
+@Component({
+  selector: 'kui-time-value',
+  templateUrl: './time-value.component.html',
+  styleUrls: ['./time-value.component.scss']
+})
+export class TimeValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @ViewChild('timeInput', {static: false}) timeInputComponent: TimeInputComponent;
+  @ViewChild('autosize', {static: false}) autosize: CdkTextareaAutosize;
+
+  @Input() displayValue?: ReadTimeValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+  date: string;
+  time: string;
+  initKnoraDate: KnoraDate; // used to display the initial value which will be an instance of KnoraDate
+  convertedControlDate: KnoraDate; // used to check the validity of the initial KnoraDate
+  updateDate: Date; // used in order to easily convert a Javascript Date object into a timestamp string using .toISOString()
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  customValidators = [];
+
+  matcher = new IntervalErrorStateMatcher();
+
+  constructor(@Inject(FormBuilder) private fb: FormBuilder, private _ngZone: NgZone) {
+    super();
+  }
+
+  standardValidatorFunc: (val: any, comment: string, commentCtrl: FormControl) => ValidatorFn
+    = (initValue: any, initComment: string, commentFormControl: FormControl): ValidatorFn => {
+    return (control: AbstractControl): { [key: string]: any } | null => {
+      
+      if(control.value.date !== null && !(control.value.date instanceof KnoraDate)){
+        this.convertedControlDate = new KnoraDate("Gregorian",
+                                                  "AD",
+                                                  control.value.date.calendarStart.year,
+                                                  control.value.date.calendarStart.month,
+                                                  control.value.date.calendarStart.day);
+      }
+
+      const invalid = (control.value !== null &&
+                       (_.isEqual(initValue.date, this.convertedControlDate) || initValue.date === control.value.date) &&
+                       (initValue.time === control.value.time || control.value.time.match(CustomRegex.TIME_REGEX) == null) &&
+                       (initComment === commentFormControl.value || (initComment === null && commentFormControl.value === '')));
+
+      return invalid ? {valueNotChanged: {value: control.value}} : null;
+    };
+  };
+
+  getInitValue(): DateTime | null {
+    if (this.displayValue !== undefined) {
+      
+      const datePipe = new DatePipe('en-US');
```

Comment:
I remember that we had strange issues with timezones when using `DatePipe`. Didn't you have some issues too?

Reply:
I haven't experienced any strange issues. The only thing is that it converts the time to the timezone of your machine.